common/lib/xmodule/xmodule/tests/test_vertical.py

```diff
@@ -202,14 +213,28 @@ def test_render_access_denied_blocks(self, has_access_error):
         self.module_system._services['user'] = StubUserService()
         self.vertical.due = datetime.now(pytz.UTC) + timedelta(days=-1)
         self.problem_block.has_access_error = has_access_error
+        self.nested_problem_block.has_access_error = has_access_error
 
         context = {'username': self.username, 'hide_access_error_blocks': True}
         html = self.module_system.render(self.vertical, STUDENT_VIEW, context).content
 
         if has_access_error:
             assert self.test_problem not in html
+            assert self.test_problem_nested not in html
         else:
             assert self.test_problem in html
+            assert self.test_problem_nested in html
+
+    @ddt.data(True, False)
+    def test_block_has_access_error(self, has_access_error):
```

Comment:
Minor: Would it be worth explicitly testing all the cases against this method? e.g. no error, error on the main block, nested error, nested nested error?

Reply:
Right now it covers: no error, error in the main block and a nested block (at the same time). Do you mean passing in two variables instead of one so I can test every combination? (Like no error, error in main block but not nested block, no error in main block but error in nested block, and errors in both blocks)?

Reply:
I think it would be nice if you can test the cases for error in the main block/not nested block, and no error in the main/with error in the nested block, since it would explicitly ensure that the check is done at both levels (main and nested). With the test case where there are errors in both the main and nested blocks, it's not clear whether it's the error in the main block or whether it's the error in the nested block that affected the result.

Reply:
I gotcha! Yes I started with these as two separate tests and Dillion had suggested combining them to re-use those setup lines, but you're right that it adds to the number of variables in the mix. I'll break them apart again

Reply:
So I would still advocate for not writing a separate test. If you want to test all of the different cases, I'd recommend doing it via ddt and setting it up as two input variables where you can then have 
True, True
True, False
False, True
False, False

Reply:
Or could simply change the format of the course. Rather than
```
        # construct module: course/sequence/vertical - problems
        #                                          \_  nested_vertical / problems
```
You could do
```
        # construct module: course/sequence/vertical - problems
        #                                 \_  vertical2/nested_vertical / problems

Reply:
Likely easier with the ddt option though. Oooh, or you could just ensure the HTML blocks continue to show up which proves it is only hiding the children blocks with the access errors. Couple of options here!

Reply:
Okay- will add another variable and expand out the ddt.data to cover them separately

Reply:
okay done!